crates/revm/src/optimism/handler_register.rs

```diff
@@ -279,14 +279,13 @@ pub fn reimburse_caller<SPEC: Spec, EXT, DB: Database>(
     context: &mut Context<EXT, DB>,
     gas: &Gas,
 ) -> Result<(), EVMError<DB::Error>> {
-    mainnet::reimburse_caller::<SPEC, EXT, DB>(context, gas)?;
+    let is_deposit = context.evm.inner.env.tx.optimism.source_hash.is_some();
 
-    if context.evm.inner.env.tx.optimism.source_hash.is_none() {
-        let caller_account = context
-            .evm
-            .inner
-            .journaled_state
-            .load_account(context.evm.inner.env.tx.caller, &mut context.evm.inner.db)?;
+    if !is_deposit {
+        mainnet::reimburse_caller::<SPEC, EXT, DB>(context, gas)?;
+    }
+
+    if SPEC::SPEC_ID.is_enabled_in(SpecId::ISTHMUS) {
```

Comment:
This doesn't quite make sense; We don't charge the operator fee on deposit transactions, so why would there be a refund?

Reply:
I'm not convinced there is any check in the operator cost func for deposit txs